Varient/SignupPage2.js

```diff
@@ -0,0 +1,182 @@
+
+    _inputCheck = () => {
+      const { firstName, lastName, isPatient, dob, gender } = this.state;
+      const requiredField = [firstName, lastName, isPatient, dob, gender];
+    
+      this.setState({ userInputError: true });
+      for (const field of requiredField) {
+        if (field === ""){
+            this.setState({  errorMessage: "Please enter all the required field" });
+            return;
+        }
+      }
+      this.setState({ userInputError: false });
+    };
+
+    _nextButtonHandler = async () => {
+      await this._inputCheck();
+      if (this.state.userInputError) {return;}
+
+      const { firstName, lastName, isPatient, dob, gender } = this.state;
+      let user = this.props.route.params.userInfo;
+      const month = dob.getMonth() + 1; //months from 1-12
+      const day = dob.getUTCDate();
+      const year = dob.getFullYear();
+
+      user["patientName"] = firstName + " " + lastName;
+      user["isPatient"] = isPatient == 1? true:false;
+      user["patientDOB"] = `${year}-${month}-${day}`;
+      user["gender"] = parseInt(gender);
+
+      this.props.navigation.navigate("SignUp3" , {userInfo: user});
+    }
+
+    render(){
+        return (
+          <View style={styles.screen}>
+            <View style={styles.greetingView}>
+              <Text style={styles.greetingText}>Enter your basic information</Text>
+              <Text style={styles.errorMessage}>
+                  {this.state.errorMessage}
+              </Text>
+
+            </View>
+            <TextInput style={styles.textInput}
+              placeholder="Your Name"
+              value={this.state.firstName}
+              onChangeText={text => this.setState({firstName:text})}
+            />
+            <TextInput style={styles.textInput}
+              placeholder="Last Name"
+              value={this.state.lastName}
+              onChangeText={text => this.setState({lastName:text})}
+            />
+
+            <TouchableOpacity
+              onPress={() => this.setState({isDatePickerVisible : !this.state.isDatePickerVisible})}
+            >
+              <Text style={styles.textInput}>{this.getDateDisplay() || "Pick your date of birth ..."}</Text>
+            </TouchableOpacity>
+
+            <TextInput style={styles.textInput}
+              placeholder="Gender"
+              value={this.state.gender}
+              onChangeText={text => this.setState({gender:text})}
+            />
+
+            <TextInput style={styles.textInput}
+              placeholder="I am ...."
+              value={this.state.isPatient}
+              onChangeText={text => this.setState({isPatient:text})}
+            />
+            <Button title="Next" onPress={() => this._nextButtonHandler()} />
+            {/* {DatePicker Modal} */}
+            {this.state.isDatePickerVisible && (
+              <DateTimePicker
+              value={this.state.dob || new Date()}
+              onChange={(event, date)=> this.setState({dob: date})}
```

Comment:
The date picker interface would not close when user presses "cancel" or "ok" on my testing device. Changing this line to this solved that part on my testing device:
```suggestion
              onChange={(event, date)=> this.setState({dob: date, isDatePickerVisible: false})}
```

Reply:
Thanks for pointing this out! The DateTimePicker component does not have an ok/cancel button on the IOS device. I've temporarily added a function to close the window once users fill out all the year, month, day. However, as this function cannot handle all the edge cases, in the future we will need to migrate to another Picker component with a native ok/cancel button for IOS